### PR TITLE
Converted app run into a webserver for repeated calls/debug

### PR DESCRIPTION
**Pull request by abhishekmaha23**
To run the code-
- Run `docker-compose up --build` as before to ensure a new image is built.

The changes are as follows-
- All current files are not copied over to the image and frozen, **live files** are **directly** used from the mapped host directory.
- A simple Flask webserver (server.py) runs on localhost:5000, and **_opening this URL in a browser will invoke a run_**. 
- Refreshing the URL in a browser will invoke a new run with the current version of the files.
- Errors are displayed in the webpage instead (through Flask's interface for stacktrace analysis).
- Logs are displayed in the console/terminal in which `docker-compose up --build` is run.
- With this change, the containers will allow live updating and changing of code and data until the Flask server running in the console is interrupted.
- To revert to the previous code, only the ENTRYPOINT to the Dockerfile need be reverted back to ['python', 'app.py']

Minor changes
- Added .idea for ignoring PyCharm IDE config files

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -18,11 +18,13 @@ COPY requirements.txt .
 RUN python -m pip install -r requirements.txt
 
 WORKDIR /app
-COPY . /app
+# COPY . /app
```
Comment (bebbi) [inline Dockerfile:21]: Would you mind directly removing unused code?

File: docker-compose.yml [hunk 2]
```diff
@@ -2,9 +2,11 @@ version: '3.4'
 
 services:
   taloc-loader:
-    image: taloc-loader
+#    image: taloc-loader
```
Comment (bebbi) [inline docker-compose.yml:5]: (same)

File: docker-compose.yml [hunk 3]
```diff
@@ -2,9 +2,11 @@ version: '3.4'
 
 services:
   taloc-loader:
-    image: taloc-loader
+#    image: taloc-loader
     build:
       context: .
       dockerfile: ./Dockerfile
+    ports:
+      - "4999:4999"
```
Comment (bebbi) [inline docker-compose.yml:10]: `5000:5000`?

**Review by bebbi — COMMENTED** [review]
Thanks for the PR! I haven't even thought about adding a webserver. How do I use it? If I open `localhost:5000`, I get the error attached below.

Also, is there a trade-off in the PR that using the VS code debugger won't be possible anymore? (which e.g. breaks before errors and allows exploring variables)

image

**Comment by abhishekmaha23** [comment]
> Thanks for the PR! I haven't even thought about adding a webserver. How do I use it? If I open `localhost:5000`, I get the error attached below.
> 
> Also, is there a trade-off in the PR that using the VS code debugger won't be possible anymore? (I like VS code breaking before errors and allowing to explore variables)
> 

I've been trying to get this to work for a bit, but it looks like since I'm using the importlib module to reimport app.py (essentially trying to rerun the whole file, triggering a run), there's no easy way to let VS Code see the variables used in any run
But I'm confident I could resolve this during the restructuring task #4 !

**Comment by bebbi** [comment]
OK I'm going to merge this one. If your approach doesn't pan out, it would still be possible to create that option (flask or simple vs code compatible run) on the docker level?

**Comment by abhishekmaha23** [comment]
I believe so! I'll test it out and check that it does

**Pull request merged**